### ISSUE TITLE
Doesn't work in Safari

### ISSUE THREAD

**Issue by arguiot** [wayou/t-rex-runner#13]
Will you ever add Safari support, for iPhone for example??

**Comment by paperclover** [comment]
i just tried it on ios 12, it does work.

**Comment by arguiot** [comment]
Yeah, Apple improved WebKit’s API on iOS which fixes the bug. I opened this issue 2 years ago, when there was an actual issue.

**Comment by null92** [comment]
Doesn't work on Safari for macOS.

**Comment by arguiot** [comment]
True, the `IS_IOS` check should include macOS as well

**Comment by chrisjreber** [comment]
Also having issues with iOS Safari, it works on iPhones but not iPads. Tested with the same version 14.4 in Simulator, multiple iPads & iPhones in Browserstack, and 3 physical devices. Not getting any click/touch events to trigger in the iPad.

**Comment by yell0wsuit** [comment]
I think the new code from Chromium has fixed that problem, but since I don't have iPads so I can't test it.
Edit: after testing, on iOS the game only works for non-Safari browsers like Chrome and Firefox. I'm investigating in [my repo](https://github.com/yell0wsuit/html5-games/tree/main/games/chrome-dino) to see what is the problem.
Edit 2: worked for Safari browsers on iOS now. But one disadvantage is that you can only touch the game area to control the dino (unlike in old code where you can press anywhere on the screen).
Edit 3: after digging deeper, the problem was ``var IS_IOS``. The old code used implementation that only applies for iOS < 12, that's why on iPad the game isn't working. I also fixed the "touch anywhere" problem.
